frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/OptionListEditor/ManualOptionsEditor/ManualOptionsEditor.tsx

```diff
@@ -0,0 +1,85 @@
+import type { IGenericEditComponent } from '@altinn/ux-editor/components/config/componentConfig';
+import type { SelectionComponentType } from '@altinn/ux-editor/types/FormComponent';
+import React, { useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  StudioCodeListEditor,
+  StudioModal,
+  StudioParagraph,
+  StudioButton,
+} from '@studio/components';
+import { PencilIcon, TrashIcon } from '@studio/icons';
+import { useOptionListEditorTexts } from '@altinn/ux-editor/components/config/editModal/EditOptions/OptionTabs/hooks';
+import type { Option } from 'app-shared/types/Option';
+import classes from './ManualOptionsEditor.module.css';
+
+type ManualOptionsEditorProps = {
+  handleDelete: () => void;
+} & Pick<IGenericEditComponent<SelectionComponentType>, 'component' | 'handleComponentChange'>;
+
+export function ManualOptionsEditor({
+  component,
+  handleComponentChange,
+  handleDelete,
+}: ManualOptionsEditorProps): React.ReactNode {
+  const { t } = useTranslation();
+  const modalRef = useRef<HTMLDialogElement>(null);
+  const editorTexts = useOptionListEditorTexts();
+
+  const handleOptionsChange = (options: Option[]) => {
+    if (component.optionsId) {
+      delete component.optionsId;
+    }
+
+    handleComponentChange({
+      ...component,
+      options,
+    });
+  };
+
+  const codeListLabels: string = component.options
+    .slice(0, 3)
+    .map((option: Option) => `${option.label}`)
+    .join(' | ');
```

Comment:
What happens if we have a very large option list? Is it a good idea to list the labels as the preview? Probably something that is discussed with @Annikenkbrathen, but I'm not sure I agree 😬 Here we should also keep in mind how the design looks for images as it is sort of the same issue.

Here is a preview example from the image component:

![Skjermbilde 2024-12-13 kl  12 30 33](https://github.com/user-attachments/assets/490d5425-d093-44ea-951d-98aa7c9b4657)

Reply:
I see now that it only uses the first three options, so I guess there will be no issue with large lists. But I still think we should rethink this. There are some cases that can mess up this view as empty strings, looong strings and as in your exemple strings that are text keys(?)

Reply:
Hmm, i was also unsure about this. It's also an option to remove it overall and only display a book icon with the name of the options list, something like this: 

![image](https://github.com/user-attachments/assets/df9b23d7-8d03-475f-bf7a-926fa136c546)

Reply:
Have you considered listing all options with `text-overflow: ellipsis`?

Reply:
Something like this?
![image](https://github.com/user-attachments/assets/9beec2f9-c825-444a-b2de-e19239f844a0)

Reply:
Yes!